server/src/main/java/org/apache/lucene/queries/BlendedTermQuery.java

```diff
@@ -133,11 +134,15 @@ protected void blend(final TermStates[] contexts, int maxDoc, IndexReader reader
                 // we need to find out the minimum sumTTF to adjust the statistics
                 // otherwise the statistics don't match
                 minSumTTF = Math.min(minSumTTF, reader.getSumTotalTermFreq(terms[i].field()));
+                minDocCount = Math.min(minDocCount, reader.getDocCount(terms[i].field()));
```

Comment:
Could we please add a small summary from the description to explain why we are doing this (for future maintainers / us), thank you

Reply:
I think I will go ahead and move the change later in the method, targeting just the fields that could go negative.

Identifying the min doc counts can still be up top, but I'll save them in an array, then consume them in the part that calls `adjustDF`.

Reply:
And I'll add a comment explaining why we're doing it of course 😀

Reply:
Done!